### PR TITLE
Add non-sequential responding and improve result of fetching

### LINKED ISSUE #5: Add and export DioAdapter without mockito
# Description

One of the implementers of the core package functionality is ought to be **DioAdapter**.

It will extend `HttpClientAdapter` from [dio](https://pub.dev/packages/dio) package and be exported in the _http_mock_adapter.dart_ file.

### PR DESCRIPTION
**Pull request by GiorgiBeriashvili**
Resolves #5

### CODE REVIEW

**Comment by nataliehippe** [comment]
Do we leave it as it is yet?

File: lib/src/interceptors/dio_interceptor.dart [hunk 1]
```diff
@@ -45,20 +44,51 @@ class DioInterceptor extends Interceptor
...
+  /// mocked request and data respectively
+  @override
+  Future<Response<dynamic>> onRequest(options) async {
+    final responseBody = history.responseBody(options);
+
+    responseBody.headers = responseBody.headers ?? {};
+
+    final headers = Headers.fromMap(responseBody.headers ?? {});
+
+    headers.set(Headers.contentTypeHeader, [Headers.jsonContentType]);
```
Comment (LukaGiorgadze) [inline lib/src/interceptors/dio_interceptor.dart:76]: I would prefer to remove extra line breaks there, like:

```
final responseBody = history.responseBody(options);
responseBody.headers = responseBody.headers ?? {};

final headers = Headers.fromMap(responseBody.headers ?? {});
headers.set(Headers.contentTypeHeader, [Headers.jsonContentType]);
```

File: test/http_mock_adapter_test.dart [hunk 2]
```diff
@@ -3,241 +3,251 @@ import 'dart:convert';
...
-  final dio = Dio();
+  Dio dio;
+
+  Map<String, dynamic> data;
   const path = 'https://example.com';
 
-  group('Interceptor', () {
-    Dio mainDio;
+  Response<dynamic> response;
+  final statusCode = 200;
```
Comment (LukaGiorgadze) [inline test/http_mock_adapter_test.dart:17]: use `const` instead of `final`

**Review by LukaGiorgadze — CHANGES_REQUESTED** [review]

**Review by nataliehippe — COMMENTED** [review]
Everything seems good. Waiting for Dato's review.

**Review by campfire5 — APPROVED** [review]

**Pull request merged**